ctl/finite_field/finite_field.hpp

```diff
@@ -57,6 +58,43 @@ std::size_t inverse( const ctl::Finite_field< N> & x, const std::size_t prime){
 	return inverse( x.value(), prime); 
 }
 
```

Comment:
CTL uses the snake_case convention, and the use of singletons is un-necessary here. 

You also don't need it: you can do something like this:
```
namespace ctl {
namespace detail {
template< std::size_t Prime>
constexpr std::array<std::size_t, Prime> inversion_table = { ctl::detail::inverse<prime>(0),  ctl::detail::inverse<prime>(1), ...,  ctl::detail::inverse<prime>(prime-1) };
} //end namespace detail
} //end namespace ctl
```

the compiler will just instantiate one (or lookup the value) when it is used..